src/mpid/ch4/generic/mpidig_send.h

```diff
@@ -90,7 +92,9 @@ static inline int MPIDI_psend_init(const void *buf,
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_PSEND_INIT);
 
     sreq = MPIDI_CH4I_am_request_create(MPIR_REQUEST_KIND__PREQUEST_SEND, 2);
-    *request = sreq;
+    if (sreq == NULL)
+        goto fn_fail;
```

Comment:
no error here?

Reply:
The outer function that calls this will do the check. Since we only return the address, there is no way to return an error code at here.

Reply:
Oh, I see what you meant.